File: Python/GoogleInterview.py
def hasPairWithSum(array, sum):
    for element in array:
        for element2 in array:
            if (element + element2 == sum):
                print(f'{element} & {element2}')
                return True
    return False

print(hasPairWithSum([1,2,3,5,9], 8))
print(hasPairWithSum([1,2,3,5,9], 13))

def hasPairWithSum2(array, sum):
    dict = {}
    for element in array:
        if element in dict:
            return True
        dict[sum - element] = True
    return False

print(hasPairWithSum2([6,4,3,2,1,7], 9))
print(hasPairWithSum2([1,2,3,5,9], 13))

File: JavaScript/GoogleInterview.js
function hasPairWithSum(arr, sum) {
    var len = arr.length;
    for (let i = 0; i < len-1; i++) {
        for (let j = 0; j < len; j++) {
            if (arr[i] + arr[j] === sum) {
                console.log(`${arr[i]} at ${i} and ${arr[j]} at ${j}`)
                return true;
            }
            
        }
    }
    return false;
}

console.log(hasPairWithSum([1,2,3,5,9], 8))
console.log(hasPairWithSum([6,4,3,2,1,7], 9))

console.log('=========================================') 

function hasPairWithSum2(arr, sum) {
    const mySet = new Set();
    const len = arr.length;
    for (let i = 0; i < len; i++) {
        if (mySet.has(arr[i])) {
            return true;
        }
        mySet.add(sum - arr[i]);
    }
    return false;
}

console.log(hasPairWithSum2([1,2,3,5,9], 8))
console.log(hasPairWithSum2([6,4,3,2,1,7], 9))
